### PR TITLE
Update `Dockerfile` and docs

### PR DESCRIPTION
**Pull request by KasiaKoz**
PR to https://github.com/arup-group/genet/pull/194 which converts all scripts to CLI.

This PR adds changes to `Dockerfile` to support this - I went with no entry point. My reasoning is that it makes it flexible to run both genet CLI and custom scripts without overriding the entry point - which is a bit of a headache if using an AWS batch job for example.

If the docker image is named something else, like 'cml-genet' then it doesn't look too crazy I think... (see the docs) - but I'm  looking for feedback on this decision.

Docker image is building: [slack build notification](https://arupconsulting.slack.com/archives/CNASXCHM2/p1698156704065569), and tests inside passing
```
$ docker run cml-genet python -m pytest -vv tests
...
=========== 1063 passed, 1 xfailed, 82 warnings in 279.00s (0:04:38) ===========
```

### CODE REVIEW

**Review by mfitz — COMMENTED** [review]
How big is the Docker image? Is it worth adding it to the GitHub release as an asset?

**Comment by KasiaKoz** [comment]
> How big is the Docker image? Is it worth adding it to the GitHub release as an asset?

@mfitz it's very big: `6.14GB` 
I have a genet image from 10 months ago that was `1.52GB`, I'm not sure what inflated the size so much (this PR: https://github.com/arup-group/genet/pull/174 brought it up to ~2gig, not sure where the rest came from, maybe we're copying jupyter notebooks and those were rerun had all the widgets saved... ? I'm only speculating here)

It would be cool to add it as an asset, but only if we trimmed the weight a bit I think

**Comment by mfitz** [comment]
> > How big is the Docker image? Is it worth adding it to the GitHub release as an asset?
> 
> @mfitz it's very big: `6.14GB` I have a genet image from 10 months ago that was `1.52GB`, I'm not sure what inflated the size so much (this PR: #174 brought it up to ~2gig, not sure where the rest came from, maybe we're copying jupyter notebooks and those were rerun had all the widgets saved... ? I'm only speculating here)
> 
> It would be cool to add it as an asset, but only if we trimmed the weight a bit I think

Hmm. When I build it locally from this branch it's 2.8 GB. Is that 6 GB IMAGE your local build, or the one in ECR?

**Review by mfitz — CHANGES_REQUESTED** [review]
The version number doesn't seem to have changed - can we do that?

**Comment by brynpickering** [comment]
It won't be the notebooks, they'd never get big enough. I would expect the image to be smaller once we reach `mamba` nirvana (it proved to be the case for PAM). Is there precedence of projects releasing docker images as GH assets @mfitz ?

File: README.md [hunk 1]
```diff
@@ -160,9 +193,14 @@ described below to verify that the code still works. All of the following comman
 directory.
 
 ### Unit tests
+To run unit tests within genet python environment:
 
     python -m pytest -vv tests
 
+and within a docker container:
+
+    docker run cml-genet python -m pytest -vv tests
```
Comment (brynpickering) [inline README.md:202]: does `docker run cml-genet pytest -vv` have the same effect?

**Comment by mfitz** [comment]
> Is there precedence of projects releasing docker images as GH assets @mfitz ?

Not in one of our own projects @brynpickering. I think we can consider publishing the docker image, though, and I can think of 3 ways to do that:

- make the GeNet ECR repo public
- publish the image to GitHub Packages
- attach the image to GitHub releases as an asset

I don't think it's a burning need, but I think it would help drive good behaviour around versioning and releases.

**Comment by KasiaKoz** [comment]
>Hmm. When I build it locally from this branch it's 2.8 GB. Is that 6 GB image your local build, or the one in ECR?

LOL! I have an ignored folder in there that has all the outputs from all the scripts after running them on londinium... I think that'll do it 😆

**Comment by brynpickering** [comment]
btw, would a publicly available genet docker image become redundant if genet was available for install directly from pip/conda?

File: README.md (hunk 1 above)
Comment (KasiaKoz) [reply README.md:202]: `docker run cml-genet pytest -vv tests` does, will change

**Comment by KasiaKoz** [comment]
the docker image serves a different purpose in my mind, you need it for containerised workflows like we use on aws---I don't know if anyone else is using it with docker (or using genet at all tbh) but personally I'm always really happy when I see a published docker image for something I want to try adding to a pipeline. Having said that, I think sharing a working `Dockerfile` is already really good.

**Comment by mfitz** [comment]
> the docker image serves a different purpose in my mind, you need it for containerised workflows like we use on aws---I don't know if anyone else is using it with docker (or using genet at all tbh) but personally I'm always really happy when I see a published docker image for something I want to try adding to a pipeline. Having said that, I think sharing a working `Dockerfile` is already really good.

Plus, the README tells people to use the Docker image, so if we're pushing that as the preferred way to use GeNet, it might be nice to have a pre-built, clearly versioned image somewhere. It's not difficult to clone the repo and build it yourself, of course, but my agenda here is making release/versioning a first-class citizen 😄 

@brynpickering If we replaced Docker as the recommended way to use GeNet with pip/conda install, the release/versioning question would take care of itself, I think, so that would solve my particular problem, but as @KasiaKoz says, we'd still need the Docker image for AWS Batch.

**Comment by KasiaKoz** [comment]
@mfitz I added notes to the changelog rather than changing the version (i didn't quote this PR though as it's a small part of another) - is that acceptable in your opinion? i don't think we want to do a release yet, we'll definitely do one after we reach the mamba nirvana

**Comment by mfitz** [comment]
> @mfitz I added notes to the changelog rather than changing the version (i didn't quote this PR though as it's a small part of another) - is that acceptable in your opinion? i don't think we want to do a release yet, we'll definitely do one after we reach the mamba nirvana

Cool @KasiaKoz 👍

**Review by mfitz — APPROVED** [review]

**Review by brynpickering — APPROVED** [review]

**Pull request merged**